SQL_DDL_Func_Mariam_Poghosyan_HW.sql

```diff
@@ -0,0 +1,196 @@
+--1
+CREATE OR REPLACE FUNCTION create_sales_revenue_by_category_qtr_view()
+RETURNS void AS $$
+DECLARE
+    view_exists BOOLEAN;
+    current_year INT := EXTRACT(YEAR FROM CURRENT_DATE);
+    current_qtr INT := EXTRACT(QUARTER FROM CURRENT_DATE);
+BEGIN
+    -- Check if the view already exists
+    SELECT EXISTS (
+        SELECT 1
+        FROM information_schema.views
+        WHERE table_name = 'sales_revenue_by_category_qtr'
+    ) INTO view_exists;
+
+    -- Drop view if it already exists to ensure rerun works
+    IF view_exists THEN
+        DROP VIEW sales_revenue_by_category_qtr;
+    END IF;
+
+    -- Create the view
+    EXECUTE format($f$
+        CREATE VIEW sales_revenue_by_category_qtr AS
+        SELECT 
+            c.name AS category,
+            SUM(p.amount) AS total_sales_revenue
+        FROM 
+            payment p
+        JOIN rental r ON p.rental_id = r.rental_id
+        JOIN inventory i ON r.inventory_id = i.inventory_id
+        JOIN film f ON i.film_id = f.film_id
+        JOIN film_category fc ON f.film_id = fc.film_id
+        JOIN category c ON fc.category_id = c.category_id
+        WHERE 
+            EXTRACT(YEAR FROM p.payment_date) = %s AND
+            EXTRACT(QUARTER FROM p.payment_date) = %s
+        GROUP BY c.name
+        HAVING SUM(p.amount) > 0
+    $f$, current_year, current_qtr);
+
+    RAISE NOTICE 'View "sales_revenue_by_category_qtr" created successfully for Year: %, Quarter: %', current_year, current_qtr;
+END;
+$$ LANGUAGE plpgsql;
+
+
+SELECT create_sales_revenue_by_category_qtr_view();
+
+
+--2
+CREATE OR REPLACE FUNCTION get_sales_revenue_by_category_qtr(qtr INT, yr INT)
+RETURNS TABLE (
+    category TEXT,
+    total_sales_revenue NUMERIC
+) AS $$
+    SELECT 
+        c.name AS category,
+        SUM(p.amount) AS total_sales_revenue
+    FROM 
+        payment p
+    JOIN rental r ON p.rental_id = r.rental_id
+    JOIN inventory i ON r.inventory_id = i.inventory_id
+    JOIN film f ON i.film_id = f.film_id
+    JOIN film_category fc ON f.film_id = fc.film_id
+    JOIN category c ON fc.category_id = c.category_id
+    WHERE 
+        EXTRACT(YEAR FROM p.payment_date) = yr AND
+        EXTRACT(QUARTER FROM p.payment_date) = qtr
+    GROUP BY c.name
+    HAVING SUM(p.amount) > 0
+$$ LANGUAGE sql STABLE;
+
+SELECT * FROM get_sales_revenue_by_category_qtr(2, 2024);
+
+--3

```

Comment:
Your function returns an incorrect result.
It should be only one film based of some criteria for each country. But your function returns a lot of rows.
![image](https://github.com/user-attachments/assets/0df73fb0-6816-4783-b4f5-1020f170d5e7)